test/parser/postgresparser_test.cpp

```diff
@@ -724,5 +724,23 @@ TEST_F(PostgresParserTests, CreateDbTest) {
   delete stmt_list;
 }
 
+TEST_F(PostgresParserTests, CreatePGSQLFuncTest) {
+  std::string query = "CREATE OR REPLACE FUNCTION increment(i integer) RETURNS integer AS $$   BEGIN  RET  j;  END; $$ LANGUAGE plpgsql;";
+
+  auto parser = parser::PostgresParser::GetInstance();
+  auto stmt_list = parser.BuildParseTree(query).release();
+  EXPECT_TRUE(stmt_list->is_valid);
+  delete stmt_list;
+}
+
+TEST_F(PostgresParserTests, CreateCFuncTest) {
+  std::string query = "CREATE FUNCTION c_overpaid(integer, integer) RETURNS boolean AS 'DIRECTORY/funcs', 'c_overpaid' LANGUAGE C STRICT;";
+  auto parser = parser::PostgresParser::GetInstance();
+  auto stmt_list = parser.BuildParseTree(query).release();
+  EXPECT_TRUE(stmt_list->is_valid);
+  delete stmt_list;
+}
+
+
 }  // End test namespace
```

Comment:
Other useful tests might be:
1) check if we can handle CREATE FUNCTION statements when there already exists a function with that name
2) implement more complex functions to ensure the parser can handle specific tables and attributes